de.fhdo.lemma.eclipse.ui/src/de/fhdo/lemma/eclipse/ui/specify_url_dialog/SpecifyUrlDialog.xtend

```diff
@@ -0,0 +1,274 @@
+            // Directory standard selection set default to workspace
+            val dirDialog = new DirectoryDialog(shell);
+            dirDialog.setText("Select your target directory")
+            dirDialog.filterPath = ResourcesPlugin.getWorkspace().getRoot().getLocation().toString()
+            val selectedDir = dirDialog.open()
+            txtTargetLocation.text = selectedDir
+        ]))
+    }
+
+    private def createDataModelName(Composite container) {
+        val lblDataModelName = new Label(container, SWT.NONE)
+        lblDataModelName.setText("Data Model Name: ")
+
+        val dataDataModelName = new GridData()
+        dataDataModelName.grabExcessHorizontalSpace = true
+        dataDataModelName.horizontalAlignment = GridData.FILL
+        dataDataModelName.horizontalSpan = 3
+
+        txtDataModelName = new Text(container, SWT.SINGLE)
+        txtDataModelName.message = "MyServiceData"
+        txtDataModelName.setLayoutData(dataDataModelName)
+    }
+
+    private def createServiceModelName(Composite container) {
+        val lblServiceModelName = new Label(container, SWT.NONE)
+        lblServiceModelName.setText("Service Model Name: ")
+
+        val dataServiceModelName = new GridData()
+        dataServiceModelName.grabExcessHorizontalSpace = true
+        dataServiceModelName.horizontalAlignment = GridData.FILL
+        dataServiceModelName.horizontalSpan = 3
+
+        txtServiceModelName = new Text(container, SWT.SINGLE)
+        txtServiceModelName.message = "MyServiceService"
+        txtServiceModelName.setLayoutData(dataServiceModelName)
+    }
+    private def createServicePrefix(Composite container) {
+        val lblServicePrefix = new Label(container, SWT.NONE)
+        lblServicePrefix.setText("Service Model Prefix: ")
+
+        val dataServicePrefix = new GridData()
+        dataServicePrefix.grabExcessHorizontalSpace = true
+        dataServicePrefix.horizontalAlignment = GridData.FILL
+        dataServicePrefix.horizontalSpan = 3
+
+        txtServicePrefix = new Text(container, SWT.SINGLE)
+        txtServicePrefix.message = "de.example"
+        txtServicePrefix.setLayoutData(dataServicePrefix)
+    }
+
+    private def createTechnologyModelName(Composite container) {
+        val lblTechnologyModelName = new Label(container, SWT.NONE)
+        lblTechnologyModelName.setText("Technology Model Name: ")
+
+        val dataTechnologyModelName = new GridData()
+        dataTechnologyModelName.grabExcessHorizontalSpace = true
+        dataTechnologyModelName.horizontalAlignment = GridData.FILL
+        dataTechnologyModelName.horizontalSpan = 3
+
+        txtTechnologyModelName = new Text(container, SWT.SINGLE)
+        txtTechnologyModelName.message = "OpenApi"
+        txtTechnologyModelName.setLayoutData(dataTechnologyModelName)
+    }
+
+    /**
+     * Stores the textfield values to actual fields.
+     * Is called when a button in the dialog is pressed.
+     */
+    private def saveInput() {
```

Comment:
Given from the comment and what the method does, its name is misleading/too generic.